.azure-pipelines/templates/jobs/standard-tests-jobs.yml

```diff
@@ -4,26 +4,24 @@ jobs:
       PYTHON_VERSION: 3.12
     strategy:
       matrix:
-        macos-py38-cover:
-          IMAGE_NAME: macOS-12
-          PYTHON_VERSION: 3.8
+        macos-py312-cover:
+          IMAGE_NAME: macOS-15
           TOXENV: cover
           # As of pip 23.1.0, builds started failing on macOS unless this flag was set.
           # See https://github.com/certbot/certbot/pull/9717#issuecomment-1610861794.
           PIP_USE_PEP517: "true"
         macos-cover:
-          IMAGE_NAME: macOS-13
+          IMAGE_NAME: macOS-15
           TOXENV: cover
           # See explanation under macos-py38-cover.
           PIP_USE_PEP517: "true"
         linux-oldest:
           IMAGE_NAME: ubuntu-22.04
           PYTHON_VERSION: 3.8
           TOXENV: oldest
-        linux-py38:
+        linux-py312:
           IMAGE_NAME: ubuntu-22.04
-          PYTHON_VERSION: 3.8
-          TOXENV: py38
+          TOXENV: py312
         linux-cover:
           IMAGE_NAME: ubuntu-22.04
           TOXENV: cover
```

Comment:
i believe we're using the same pattern here testing on linux with the oldest and newest python we support, except we're only measuring/testing coverage once. as of now in this PR, these tests aren't identical, but i believe `linux-cover` is a superset of the tests run in `linux-py312`